### ISSUE TITLE
Build badge pages and enable claim codes for Hive Toronto badges

### ISSUE THREAD

**Issue by OpenMatt** [MozillaFoundation/Mozilla-Learning#11]
As per: https://github.com/MozillaFoundation/plan/issues/206#issuecomment-73714841
* We need to build the pages @iamjessklein is designing here: https://github.com/MozillaFoundation/Mozilla-Learning/issues/10

* We will look to re-purpose this existing template:
https://webmaker.org/badges/hive-community-member

* These templates will need to be adapted to work with claim codes. The user will enter a claim code, and then be issued the badge. Note that claims codes are the *only* way the user can earn the badge. 

* @ScottDowne : you'll need to do some research on how claim codes work. 

@simonwex : My understanding is that @ScottDowne is going to help build these pages and enable claim codes to work.

**Comment by OpenMatt** [comment]
* Main thing we need to synch up on, to ensure design and dev are on the same page: 
how do claim codes work? And: do we need to tweak the current badge landing page design to support the claim code workflow we want?

**Comment by smithisgeneric** [comment]
This is a starting sketch from a conversation @OpenMatt and I had 

![hivebadgeflow](https://cloud.githubusercontent.com/assets/6931814/6132068/31b87cc0-b11e-11e4-9869-cd5e81bdfeb5.jpg)

**Comment by OpenMatt** [comment]
@ as per conversation with @ScottDowne : this work doesn't necessarily have to be gated on design. We can take the existing page template, fork it, throw in a janky "enter claim code" field, and start testing and optimizing. Then update with prettier design when it's done.

**Comment by OpenMatt** [comment]
@ScottDowne does this make sense? And are you able to start work on this without blocking on design?

**Comment by ScottDowne** [comment]
@OpenMatt Question, the pages in question don't yet exist as a badge yet, correct?

The templates just load any existing badge, it seems.

**Comment by ScottDowne** [comment]
Why don't we just add a claim code input for all badges? That way we continue to only have 1 template.

**Comment by ScottDowne** [comment]
I'm seeing some already similar implementation on webmaker.org's badge pages.

Example, if you click apply while logged in from here: https://webmaker.org/badges/hive-community-member

You get this:

![claim](https://cloud.githubusercontent.com/assets/197334/6176818/c7d42e40-b2cf-11e4-9287-86acfd0432d6.png)

And once you claim it, you see this:
![success](https://cloud.githubusercontent.com/assets/197334/6176823/cb67ca80-b2cf-11e4-98c0-590b518be1f9.png)

**Comment by OpenMatt** [comment]
@ScottDowne : correct: the badges don't exist yet. The graphic design work on them is still in progress. As per: https://github.com/MozillaFoundation/plan/issues/206

* Good point about only requiring one template. We should definitely try to do it that way.

**Comment by hannahkane** [comment]
Late to the party (story of my life), but I believe we want to create a separate use case for these badges because we *don't* want people to apply for them the "normal" way. We want the claim code to be the *only* method for earning these badges.

@ScottDowne - is that possible?

Also,@simonwex , is there a separate ticket for fixing the apply/login bug? Can we link that here as a dependency?

**Comment by smithisgeneric** [comment]
I updated the copy for the 3 badge pages and have requested review from @OpenMatt @iamjessklein.  

The copy deck contains the Wordpress landing page, plus the 3 badge pages which which I requested simultaneous review on [Issue 12] (https://github.com/MozillaFoundation/Mozilla-Learning/issues/12).

**Comment by smithisgeneric** [comment]
Responding the @hannahkane on the fields for earning the badge.

- Keeping it to claims code only is preferred because there aren't the people resources to assess the makes and projects as evidence to earn the badges; we'd like the educators to do the assessment when they utilize the learning activities (i.e., they run a take-up discussion with youth following an activity and issue claims codes)  

The risk of having all fields is that a big backlog of applications for badges with no staff member available to assess the projects.

**Comment by ScottDowne** [comment]
@hannahkane yes, it is possible to restrict the apply methods on a per badge basis.

If possible I'll see about storing that info in the badge itself and the page can be smart enough to know what fields to display based on the data stored in the badge. Not sure if that's possible.

If it's not, I know I can just as easily, although not as elegantly, tell the page itself about our new badges.

"There's a related bug re: the Apply button and logging in" yes, those are fixed and are landing today.

**Comment by davidascher** [comment]
I can understand that we want a claim-only flow for the toronto event, but it seems to me we likely want to allow for a possible scenario where people get the same badges in the future by providing evidence.

How about if we make a URL to the badge display page that only shows the claimcode path, and use that in the hive wordpress thing, but still make it so that if people got to those badges through "normal" discovery (via the as-yet-nonexistent) badge directory, then they could submit evidence?

For @ScottDowne  that would mean hiding part of the current claiming UI based on a query parameter for example.  For @smithisgeneric I think it would mean tweaking the URLs in the page.

I think the question is for @hannahkane given the larger picture and longer-term badges roadmap.

(Trying to make sure that we don't build too many special cases in specific badges that we then regret, and instead isolate the 'specialness' to the event in question, not to the badge itself).

I could be misunderstanding the requirements, would love a reaction or two.

**Comment by ScottDowne** [comment]
A url query would work and I like that solution.

One potential problem with this that I want to point out, if people are sharing urls that they don't understand, and then in the future try to apply for it and somehow use the wrong url, they won't be able to and won't know why they cannot apply for it.

If there was a special case, it would be trivial to turn on and off. It's super gross though and not something I am happy about.

If the apply methods were stored as metadata on the badge itself, turning it on and off would not be trivial. So far I am told updating badge data requires sql commands directly on the data base.

**Comment by hannahkane** [comment]
I think changing the query parameters makes sense, because I imagine these privacy badges will be included in the suite of web lit badges we offer on the Clubs platform, where they would be applied for in the "normal" way. So having both options seems ideal.

I echo @ScottDowne's concern, though. I hope this is a very rare use case.

**Comment by ScottDowne** [comment]
All done

https://webmaker.org/en-US/badges/privacy-coach
https://webmaker.org/en-US/badges/ip-address-tracer
https://webmaker.org/en-US/badges/data-trail-timeline

I ended up hard coding the badges we want to not display an apply button for and decided to not use the query string based on advice from Kate.

Closing this.

Any issues found should be filed as new tickets.

**Comment by toolness** [comment]
(Sorry, the aforementioned commit was not actually referring to this issue--it was referring to toolness/mozilla-learning-html#11 and we later imported that code into this repository.)

**Comment by toolness** [comment]
<a href="https://github.com/mozilla/learning-networks/issues/1"><img src="https://raw.githubusercontent.com/toolness/learning-issue-migrator/master/img/migrated.png"></a>
This issue has been moved to https://github.com/mozilla/learning-networks/issues/1.
Please don't comment on it here.
